src/Stratis.Bitcoin/P2P/PeerConnectorConnect.cs

```diff
@@ -70,8 +70,41 @@ public override void OnStartConnect()
 
                 PeerAddress peerAddress = this.peerAddressManager.FindPeer(ipEndpoint);
                 if (peerAddress != null && !this.IsPeerConnected(peerAddress.Endpoint))
-                    await ConnectAsync(peerAddress).ConfigureAwait(false);
+                {
+                    // Nodes disallow connection to peers in same range to prevent sybil attacks.     
+                    if (ipEndpoint.Address.IsLocal())
+                    {
+                        // Local peer: filtering is disabled unless explicitly set to true.
+                        if (this.ConnectionSettings.IpRangeFiltering == true)
+                        {
+                            if (this.PeerIsPartOfExistingGroupOfConnectedEndpoints(peerAddress))
+                            {
+                                continue;
+                            }
+                        }
+                        await this.ConnectAsync(peerAddress).ConfigureAwait(false);
```

Comment:
Code style and also don't repeat below the same code. Just greater variable weather to connect or not based on whether the address is local or not and whether it is in the same group.

Reply:
This has been refactored completely.